pkg/api/handlers/compat/containers_top.go

```diff
@@ -52,13 +52,19 @@ func TopContainer(w http.ResponseWriter, r *http.Request) {
 
 	encoder := json.NewEncoder(w)
 
+	args := query.PsArgs
+	if len(args) == 1 {
```

Comment:
I think we should also guard this with `utils.IsLibpodRequest(r)` and `utils.SupportedVersion(r, "< 4.8.0")`, then we only do this for the old libpod API.

The docker API will only ever send us a single arg which will already be shlex.Split() in libpod backend so this should be fine.